### PR TITLE
New tool for quantifying artifacts (with code review changes)

### PR DESCRIPTION
**Pull request by mattsooknah**
New tool for quantifying substitution artifacts from sample/library prep. Loosely based on CollectOxoGMetrics.

Depends on https://github.com/samtools/htsjdk/pull/206 (that's why the build is failing right now)

### CODE REVIEW

**Comment by nh13** [comment]
Could you change the name of the test cases directory to match the names and hierarchy of other tests (if I recall correct there is not other TestCases subdirectory)?

File: src/java/picard/analysis/CompareMetrics.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    protected int doWork() {
+        IOUtil.assertFilesAreReadable(metricsFiles);
+        final MetricsFile<?, ?> metricsA = new MetricsFile();
+        final MetricsFile<?, ?> metricsB = new MetricsFile();
+        try {
+            metricsA.read(new FileReader(metricsFiles.get(0)));
+            metricsB.read(new FileReader(metricsFiles.get(1)));
+            final boolean areEqual = metricsA.areMetricsEqual(metricsB) && metricsA.areHistogramsEqual(metricsB);
+            final String status = areEqual ? "EQUAL" : "NOT EQUAL";
+            System.out.println("Files " + metricsFiles.get(0) + " and " + metricsFiles.get(1) + "are " + status);
```
Comment (mcovarr) [inline src/java/picard/analysis/CompareMetrics.java]: Should this be writing to standard output?

File: src/java/picard/analysis/CollectOxoGMetrics.java [hunk 2]
```diff
@@ -29,10 +30,11 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashSet;
-import java.util.LinkedList;
 import java.util.List;
 import java.util.Set;
 
+import static htsjdk.samtools.util.CodeUtil.nvl;
+import static htsjdk.samtools.util.SequenceUtil.generateAllKmers;
```
Comment (mcovarr) [inline src/java/picard/analysis/CollectOxoGMetrics.java]: These new imports don't appear to be used.

**Comment by mattsooknah** [comment]
@nh13 what do you mean? TestCases is a subdirectory of the test directory containing the expected output metrics (the input files are one level above that). This is to distinguish between input/output and make things a little cleaner. I could name it to "ExpectedMetricsOutput" if it's too ambiguous.

File: src/java/picard/analysis/CompareMetrics.java (hunk 1 above)
Comment (mattsooknah) [inline src/java/picard/analysis/CompareMetrics.java]: The existing convention seems to be that for large CLPs, we use logging; but for small CLPs, we often just write everything to stdout. If people feel strongly I can change it, but there are lots of ```println```'s in existing CLPs.

File: src/java/picard/analysis/CollectOxoGMetrics.java (hunk 2 above)
Comment (mattsooknah) [inline src/java/picard/analysis/CollectOxoGMetrics.java]: They are used, although it's not visible in the diff.

File: src/java/picard/analysis/CollectOxoGMetrics.java [hunk 3]
```diff
@@ -263,11 +233,14 @@ protected int doWork() {
         }
         iterator.setEmitUncoveredLoci(false);
         iterator.setMappingQualityScoreCutoff(MINIMUM_MAPPING_QUALITY);
-        iterator.setSamFilters(Arrays.asList(
-                new NotPrimaryAlignmentFilter(),
-                new DuplicateReadFilter(),
-                new InsertSizeFilter(MINIMUM_INSERT_SIZE, MAXIMUM_INSERT_SIZE)
-        ));
+
+        final ArrayList<SamRecordFilter> filters = new ArrayList<SamRecordFilter>();
```
Comment (mcovarr) [inline src/java/picard/analysis/CollectOxoGMetrics.java]: `List<SamRecordFilter>` on LHS?

File: src/java/picard/analysis/CompareMetrics.java (hunk 1 above)
Comment (mcovarr) [inline src/java/picard/analysis/CompareMetrics.java]: I'd vote for using logs consistently going forward, but not for going back and fixing existing CLPs.  :smile:

**Comment by nh13** [comment]
@mattsooknah do other classes have a ```TestCases``` sub-directory?  If so, the all good, if not, then change it.  I am trying to enforce uniformity.

**Comment by mattsooknah** [comment]
Verbal :+1: from Nils, just waiting for Travis now.

**Comment by yfarjoun** [comment]
Awesome!

**Pull request merged**